### DISCUSSION TITLE
TypeError: globalThis.__sbPostVisit is not a function

### DISCUSSION THREAD

**Discussion by matttk** [storybookjs/storybook#29761] [Help] [answered]
### Summary

I don't have a repro right now (sorry), and all I can really say is that I did a big re-write to replace our datepicker components to use [mantine.dev](https://mantine.dev), instead of [react-dates](https://www.npmjs.com/package/react-dates), and then I started noticing a new test flake.

> FAIL app/components/BuildingBlocks/Dates/DateInput/DateInput.stories.tsx
> Generic/Dates/DateInput › WithoutFutureDates › play-test
> 
> TypeError: globalThis.__sbPostVisit is not a function

I can't find anything about what **sbPostVisit** is, but I assume sb stands for storybook...

It's happening at random in any component that uses a DatePicker in our project. I have the latest version of Storybook, 8.4.6.

Update 1: Maybe related to this issue? https://github.com/storybookjs/test-runner/issues/138

Update 2: I've determined they're all actual test fails, only that I don't get the real error. But everywhere where I get `globalThis.__sbPostVisit is not a function` corresponds to a failing test, which I can find by looking in Storybook locally (not sure how they were green in the CI before but maybe I updated something else, which broke the tests).

**Comment by hasanayan** [comment]
We, too, started noticing this yesterday. It seems there are actual test failures but storybook is unable to provide a meaningful error message or the origin of the error. I get that error locally on my mac as well

**Comment by hasanayan** [comment, marked as answer]
@matttk I think `"@storybook/test-runner": "0.20.1",` fixed it for us

**Reply by matttk** [reply]
Ah, very cool. I'll update it today, but sounds like there was a problem and it's been fixed. I also never thought to look into the `test-runner` release notes directly, which in retrospect was a bit silly.
